src/app/result/page.tsx

```diff
@@ -1,37 +1,45 @@
 'use client';
 
-import { ShareButton } from '@/components/ui/ShareButton';
+import React, { Suspense } from 'react';
 import { useSearchParams } from 'next/navigation';
-import { useEffect, useState } from 'react';
-import Loading from './loading';
-
-export default function Page() {
-  const [loading, setLoading] = useState(true);
-  const battlePowerResult = useSearchParams();
-  const score = battlePowerResult ? battlePowerResult.get('score') : null;
+import { ShareButton } from '@/components/ui/ShareButton';
+import { Canvas } from '@react-three/fiber';
+import { Html } from '@react-three/drei';
+import styles from '@/styles/scouterText.module.css';
 
+// SearchParamsを取得するコンポーネント
+function ResultContent() {
+  const searchParams = useSearchParams();
+  const score = searchParams.get('score');
+  
   // TODO: 以下のエラーが発生した時、ユーザーのトップページに戻すための何かしらの処理を考える
   const error =
     score === null || score === '' || Number.isNaN(Number(score))
       ? 'スカウターの測定に失敗しました。もう一度お試しください。'
       : '';
 
-  // NOTE: ローディング画面を一定時間表示するための一時的な処理
-  useEffect(() => {
-    const timer = setTimeout(() => setLoading(false), 4000);
-    return () => clearTimeout(timer);
-  }, []);
-
-  if (loading) {
-    return Loading();
-  }
-
   return (
-    <>
+    <Html fullscreen>
       <h2>結果</h2>
       {!error && <h3>{score}</h3>}
       {error && <div className="text-sm text-red-500">{error}</div>}
       <ShareButton tweetText={String(score)} />
```

Comment:
今回の範囲じゃなくてすみませんmm
シェアボタンもエラーの有無によって出しわけした方が良さそうですねmm
```suggestion
      {!error && <ShareButton tweetText={String(score)} />}
```